src/main/java/pro/quicksense/service/PaymentService.java

```diff
@@ -0,0 +1,237 @@
+package pro.quicksense.service;
+
+import com.alibaba.fastjson.JSON;
+import com.alibaba.fastjson.JSONArray;
+import com.alibaba.fastjson.JSONObject;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import pro.quicksense.annotation.AirwallexRequest;
+import pro.quicksense.common.AirwallexConstant;
+import pro.quicksense.util.TimezoneUtil;
+
+import java.text.ParseException;
+import java.util.Date;
+
+import static pro.quicksense.common.AirwallexConstant.*;
+
+@Service
+public class PaymentService {
+
+    /**
+     * As a client to Airwallex, all the requests should be sent with a token returned from authentication.
+     */
+    private static String AIRWALLEX_TOKEN;
+
+    /**
+     * The timestamp the token would expire at.
+     */
+    private static String AIRWALLEX_TOKEN_EXPIRES_AT;
+
+    /**
+     * Request for Airwallex authentication, Airwallex would return a time-sensitive token,
+     * which should be included in all the other requests to Airwallex.
+     */
+    public void authByAirwallex() throws UnirestException, ParseException {
+        Date currentDate = new Date();
+        // TODO Check the correctness of the date.
+        if (StringUtils.isNotEmpty(AIRWALLEX_TOKEN_EXPIRES_AT)
+                && currentDate.before(TimezoneUtil.convertTimeStringToDateObject(AIRWALLEX_TOKEN_EXPIRES_AT))) {
+            return;
+        }
+        HttpResponse<String> response = Unirest.post(AirwallexConstant.API_AUTHENTICATION_LOGIN)
+                .header("Content-Type", "application/json")
+                // TODO Replace the 'CLIENT_ID_TEST' and 'API_KEY_TEST' with configurations for production.
+                .header("x-client-id", AirwallexConstant.CLIENT_ID_TEST)
+                .header("x-api-key", AirwallexConstant.API_KEY_TEST)
+                .body("{}")
+                .asString();
+        JSONObject jsonObject = JSON.parseObject(response.getBody());
+        AIRWALLEX_TOKEN = (String) jsonObject.get(AirwallexConstant.TOKEN);
```

Comment:
```suggestion
        JSONObject jsonObject = Unirest.post(API_AUTHENTICATION_LOGIN)
                .header("Content-Type", "application/json")
                // TODO Replace the 'CLIENT_ID_TEST' and 'API_KEY_TEST' with configurations for production.
                .header("x-client-id", CLIENT_ID_TEST)
                .header("x-api-key", API_KEY_TEST)
                .body("{}")
                .asJson()
                .getBody()
                .getObject();

        jsonObject.getJSONObject(AirwallexConstant.TOKEN);
```

Reply:
Reference: https://kong.github.io/unirest-java/#responses

Reply:
The code you provided complains -> An error occurred: Type definition error: [simple type, class org.json.JSONObject]